instantlaunches/mgmt.go

```diff
@@ -0,0 +1,180 @@
+	VALUES ( $1, ( SELECT u.id FROM users u WHERE u.username = $2 ) )
+	RETURNING id, quick_launch_id, added_by, added_on;
+`
+
+// AddInstantLaunch registers a new instant launch in the database.
+func (a *App) AddInstantLaunch(quickLaunchID, username string) (*InstantLaunch, error) {
+	newvalues := &InstantLaunch{}
+	err := a.DB.QueryRowx(addInstantLaunchQuery, quickLaunchID, username).Scan(&newvalues.ID,
+		&newvalues.QuickLaunchID,
+		&newvalues.AddedBy,
+		&newvalues.AddedOn,
+	)
+	return newvalues, err
+}
+
+// AddInstantLaunchHandler is the HTTP handler for adding a new instant launch.
+func (a *App) AddInstantLaunchHandler(c echo.Context) error {
+	il, err := NewInstantLaunchFromJSON(c.Request().Body)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "cannot parse JSON")
+	}
+
+	if il.AddedBy == "" {
+		return echo.NewHTTPError(http.StatusBadRequest, "username was not set")
+	}
+
+	if !strings.HasSuffix(il.AddedBy, a.UserSuffix) {
+		il.AddedBy = fmt.Sprintf("%s%s", il.AddedBy, a.UserSuffix)
+	}
+
+	newil, err := a.AddInstantLaunch(il.QuickLaunchID, il.AddedBy)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return echo.NewHTTPError(http.StatusNotFound, err.Error())
+		}
+		return err
+	}
+
+	return c.JSON(http.StatusOK, newil)
+}
+
+const getInstantLaunchQuery = `
+	SELECT i.id, i.quick_launch_id, i.added_by, i.added_on
+	  FROM instant_launches i
+	 WHERE i.id = $1;
+`
+
+// GetInstantLaunch returns a stored instant launch by ID.
+func (a *App) GetInstantLaunch(id string) (*InstantLaunch, error) {
+	il := &InstantLaunch{}
+	err := a.DB.QueryRowx(getInstantLaunchQuery, id).Scan(&il.ID,
+		&il.QuickLaunchID,
+		&il.AddedBy,
+		&il.AddedOn,
+	)
+	return il, err
+}
+
+// GetInstantLaunchHandler is the HTTP handler for getting a specific Instant Launch
+// by its UUID.
+func (a *App) GetInstantLaunchHandler(c echo.Context) error {
+	id := c.Param("id")
+	if id == "" {
+		return echo.NewHTTPError(http.StatusBadRequest, "id is missing")
+	}
+
+	il, err := a.GetInstantLaunch(id)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return echo.NewHTTPError(http.StatusNotFound, err.Error())
+		}
+		return err
+	}
+
+	return c.JSON(http.StatusOK, il)
+
+}
+
+const updateInstantLaunchQuery = `
+	UPDATE ONLY instant_launches
+			SET quick_launch_id = $1
+		  WHERE id = $2
+	  RETURNING id, quick_launch_id, added_by, added_by;
+`
+
+// UpdateInstantLaunch updates a stored instant launch with new values.
+func (a *App) UpdateInstantLaunch(id, quickLaunchID string) (*InstantLaunch, error) {
+	il := &InstantLaunch{}
+	err := a.DB.QueryRowx(updateInstantLaunchQuery, quickLaunchID, id).Scan(&il.ID,
+		&il.QuickLaunchID,
+		&il.AddedBy,
+		&il.AddedOn,
+	)
+	return il, err
+}
+
+// UpdateInstantLaunchHandler is the HTTP handler for updating an instant launch.
+func (a *App) UpdateInstantLaunchHandler(c echo.Context) error {
+	id := c.Param("id")
+	if id == "" {
+		return echo.NewHTTPError(http.StatusNotFound, "id is missing")
```

Comment:
This is a close call, but I think I'd go with `http.StatusBadRequest` here instead. Of course, since `id` is the last element in the URL path, the point might be moot. Feel free to change this or not. 🙂

Reply:
Yeah, I've been going back and forth on that. Not sure what the right call is. I'll leave it for now.